src/merge-fields/mergeField.js

```diff
@@ -0,0 +1,187 @@
+
+
+  // We want to add the incoming subfields without punctuation, and add puctuation later on.
+  // (Cloning is harmless, but probably not needed.)
+  // NEW: we also drag the normalized version along. It is needed for the merge-or-add decision
+  const normalizedSourceField = cloneAndNormalizeFieldForComparison(sourceField); // This is for comparison
+  const strippedSourceField = cloneAndRemovePunctuation(sourceField); // This is for adding subfields
+
+  //nvdebug(`  MERGING SUBFIELDS OF '${fieldToString(sourceField)}' (original)`, debugDev);
+  //nvdebug(`  MERGING SUBFIELDS OF '${fieldToString(normalizedSourceField)}' (comparison)`, debugDev);
+  nvdebug(`  MERGING SUBFIELDS OF '${fieldToString(strippedSourceField)}' (merge/add)`, debugDev);
+
+  sourceField.subfields.forEach((originalSubfield, index) => {
+  //strippedSourceField.subfields.forEach((subfieldForMergeOrAdd, index) => {
+    const normalizedSubfield = normalizedSourceField.subfields[index];
+    const punctlessSubfield = strippedSourceField.subfields[index];
+    const originalBaseValue = fieldToString(baseField);
+    nvdebug(`  TRYING TO MERGE SUBFIELD '${subfieldToString(originalSubfield)}' TO '${originalBaseValue}'`, debugDev);
+
+    const subfieldData = {'tag': sourceField.tag, 'code': originalSubfield.code, 'originalValue': originalSubfield.value, 'normalizedValue': normalizedSubfield.value, 'punctuationlessValue': punctlessSubfield.value};
+
+    mergeOrAddSubfield(baseField, subfieldData, candFieldPairs880); // candSubfield);
+    const newValue = fieldToString(baseField);
+    if (originalBaseValue !== newValue) { // eslint-disable-line functional/no-conditional-statements
+      nvdebug(`   SUBFIELD MERGE RESULT: '${newValue}'`, debugDev);
+      //debug(`   TODO: sort subfields, handle punctuation...`);
+    }
+    //else { debugDev(`  mergeOrAddSubfield() did not add '‡${fieldToString(subfieldForMergeOrAdd)}' to '${originalValue}'`); }
+
+  });
+}
+
+
+function skipMergeField(baseRecord, sourceField, config) {
+  if (!mergableTag(sourceField.tag, config)) {
+    nvdebug(`skipMergeField(): field '${fieldToString(sourceField)}' listed as skippable!`, debugDev);
+    return true;
+  }
+
+  // Skip duplicate field:
+  if (baseRecord.fields.some(baseField => !baseField.mergeCandidate && fieldsAreIdentical(sourceField, baseField))) {
+    nvdebug(`skipMergeField(): field '${fieldToString(sourceField)}' already exists! No merge required!`, debugDev);
+    sourceField.deleted = 1; // eslint-disable-line functional/immutable-data
+    return true;
+  }
+
+  return false;
+}
+
+function sourceRecordIsBetter(baseField, sourceField) {
+  if (!baseField.subfields) {
+    return;
+  }
+  // MELINDA-8978: prefer Asteri version
+  if (isAsteriField(sourceField) && !isAsteriField(baseField)) {
+    return 1;
+  }
+
+  function isAsteriField(field) {
+    if (field.subfields.some(sf => sf.code === '0' && sf.value.match(/^\((?:FI-ASTERI-N|FIN11)\)[0-9]{9}$/u))) {
```

Comment:
We'll probably want to handle also Asteri-W fields as preferable to non-Asteri fields. What to do with Asteri-N vs Asteri-W fields is more complicated.

Reply:
Added.